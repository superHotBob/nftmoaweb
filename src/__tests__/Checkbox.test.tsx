import React from "react";
import { shallow, ShallowWrapper } from "enzyme";
import { create, ReactTestRenderer } from "react-test-renderer";

import CheckBox, { CheckBoxProp } from "../components/Test/Checkbox";

let wrapper: ShallowWrapper<CheckBoxProp>;
let snapshot: ReactTestRenderer;

beforeEach(() => {
  const checkbox = <CheckBox labelOn="On" labelOff="Off" />;

  wrapper = shallow(checkbox);
  snapshot = create(checkbox);
});

describe("<CheckBox />", () => {
  test("it matches the snapshot", () => {
    expect(snapshot.toJSON()).toMatchSnapshot();
  });

  it("체크 박스 클릭 이벤트", () => {
    expect(wrapper.text()).toEqual("Off");
    wrapper.find("input").simulate("change");
    expect(wrapper.text()).toEqual("On");
  });
});
