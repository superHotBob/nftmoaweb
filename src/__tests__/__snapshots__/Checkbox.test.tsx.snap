// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CheckBox /> it matches the snapshot 1`] = `
<label>
  <input
    checked={false}
    onChange={[Function]}
    type="checkbox"
  />
  Off
</label>
`;
